ui/lib/sync/addon/routes/secrets/destinations/destination.ts

```diff
@@ -15,9 +18,22 @@ interface RouteParams {
 
 export default class SyncSecretsDestinationsDestinationRoute extends Route {
   @service declare readonly store: Store;
+  @service declare readonly router: RouterService;
 
   model(params: RouteParams) {
     const { name, type } = params;
     return this.store.findRecord(`sync/destinations/${type}`, name);
   }
+
+  afterModel(model: SyncDestinationModel, transition: Transition) {
+    // handles the case where the user attempts to perform actions on a destination when a purge has been initiated
+    // editing is available from the list view and syncing secrets is available from the overview
+    // the list endpoint does not return the full model so we don't have access to purgeInitiatedAt to disable or hide the actions
+    // if transitioning from either of the mentioned routes and a purge has been initiated redirect to the secrets view
+    const baseRoute = 'vault.cluster.sync.secrets.destinations.destination';
+    const routes = [`${baseRoute}.edit`, `${baseRoute}.sync`];
+    if (routes.includes(transition.to.name) && model.purgeInitiatedAt) {
```

Comment:
Should we add some sort of flash message indicating why the user is not on the page they intended to visit?